pkg/extension/sumologicextension/extension.go

```diff
@@ -34,6 +35,7 @@ type sumologicExtension struct {
 	conf             *Config
 	logger           *zap.Logger
 	registrationInfo OpenRegisterResponsePayload
+	ch               chan bool
```

Comment:
```suggestion
	ch               chan struct{}
```

? And close it (guarded by `sync.Once`) as a signal instead of sending stuff down that channel?